docs/deployment/deployment-checklist.md

```diff
@@ -0,0 +1,258 @@
+    {label: 'Bitbucket', value: 'bb'},
+    ]}
+>
+
+<TabItem value='gh'>
+
+#### GitHub
+
+| Feature | Permission required |
+| --- | -------  | 
+| Create CI jobs for repositories in bulk and detect GitHub repos automatically. | Install GitHub apps.         |
+| Pull request (PR) comments. |  For GitHub Enterprise Server: Add a personal access token (PAT) with [assigned scopes](/deployment/connect-scm/#connect-to-on-premise-github-or-gitlab-orgs).          |
+| GPT-assisted triage and recommendations. | Code access. |
+
+</TabItem>
+
+<TabItem value='gl'>
+
+#### GitLab
+
+<table>
+<thead>
+<tr>
+<th>Feature</th>
+<th>Permission required</th>
+</tr>
+</thead>
+<tbody>
+<tr>
+<td>Merge request (MR) comments.</td>
+<td>Create personal access tokens.</td>
+</tr>
+<tr>
+<td rowspan="2">GPT-assisted triage and recommendations.</td>
+<td>Create personal or project-level access tokens.</td>
+</tr>
+<tr>
+<td>Code access.</td>
+</tr>
+</tbody>
+</table>
+
+
+</TabItem>
+
+<TabItem value='bb'>
+
+#### Bitbucket 
+
+| Feature  | Permission |
+| -------  | -------  | 
+| Pull request (PR) comments.  | Able to create **repository variables**. | 
+
+</TabItem>
+
+</Tabs>
+
+## Appendices
+
+### Permissions
+
+<Tabs
+    defaultValue="permissions-github"
+    values={[
+    {label: 'Permissions for GitHub', value: 'permissions-github'},
+    {label: 'Permissions for GitLab', value: 'permissions-gitlab'},
+    ]}
+>
+<TabItem value='permissions-github'>
+
+#### Permissions for GitHub
+
+This section explains Semgrep Cloud Platform permissions that are requested in two different events:
+
+* When you first sign in through GitHub.
+* When you first add, integrate, or onboard your repositories to Semgrep Cloud Platform.
+
+##### Permissions when signing in with GitHub
+
+Semgrep Cloud Platform requests the following standard permissions set by GitHub when you first sign in. However, not all permissions are used by Semgrep Cloud Platform. Read the following list to see how Semgrep Cloud Platform uses permissions when signing in:
+
+<dl>
+    <dt>Verify your GitHub identity</dt>
+    <dd>Enables Semgrep Cloud Platform to read your GitHub profile data, such as your username.</dd>
+    <dt>Know which resources you can access</dt>
+    <dd>Semgrep does not use or access any resources when first logging in. However, you can choose to share resources at a later point to add repositories into Semgrep Cloud Platform.</dd>
+    <dt>Act on your behalf</dt>
+    <dd>Enables Semgrep Cloud Platform to perform certain tasks <strong>only on resources that you choose to share with Semgrep Cloud Platform</strong>. Semgrep Cloud Platform never uses this permission and never performs any actions on your behalf, even after you have installed <code>semgrep-app</code>. See <a href ="https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/authorizing-github-apps">When does a GitHub App act on your behalf?</a> in GitHub documentation.</dd>
```

Comment:
I'm confused by the "Semgrep Cloud Platform never uses this permission" statement. We must be using it no?

Reply:
Hahaha I had the exact same question when I wrote this originally. When a user "authorizes" a github app, there are 3 permissions at the minimum that GitHub requires the user accept for ANY GitHub app. This is regardless of any actual need for those permissions. The link provided to the github docs states:

> Act on your behalf: When authorized, the application may perform tasks on GitHub on your behalf. This might include creating an issue or commenting on a pull request. For more information, see "[About GitHub Apps acting on your behalf](https://docs.github.com/en/apps/using-github-apps/authorizing-github-apps#about-github-apps-acting-on-your-behalf)."

You will notice that when a comment is posted, it is `semgrep-app` that performs the action, not the user. So `semgrep-app` is not acting on a user's behalf (or it would have used the user's account).